src/frontend/components/message.js

```diff
@@ -0,0 +1,361 @@
+			box-shadow: rgba(0, 0, 0, 0.16) 0px 1px 4px;
+			z-index: 3;
+		}
+
+		.sender .message > p::selection {
+			color: black;
+			background-color: white;
+		}
+
+		input {
+			font-family: inherit;
+		}
+
+		.sender {
+			justify-self: flex-end;
+			border-radius: 10px 0 10px 10px;
+
+			color: white;
+			background-color: rgb(54, 123, 251);
+			justify-self: flex-end;
+		}
+		.sender::after {
+			content: "";
+			position: absolute;
+			top: 0px;
+			right: -9px;
+			border-top: 10px solid rgb(54, 123, 251);
+			border-left: 0px solid transparent;
+			border-right: 10px solid transparent;
+			z-index: 3;
+		}
+		.sender::before {
+			content: "";
+			position: absolute;
+			top: -1px;
+			right: -13px;
+			border-top: 11px solid rgb(209 209 209 / 34%);
+			border-left: 0px solid transparent;
+			border-right: 12px solid transparent;
+			filter: blur(0.8px);
+			z-index: 2;
+		}
+
+		.sender a:link {
+			color: black;
+		}
+
+		.sender a:visited {
+			color: black;
+		}
+
+		.sender a:hover {
+			color: white;
+		}
+
+		.receiver {
+			justify-self: flex-start;
+			border-radius: 0 10px 10px 10px;
+
+			color: black;
+			background-color: white;
+		}
+		.receiver::after {
+			content: "";
+			position: absolute;
+			top: 0px;
+			left: -9px;
+			border-top: 10px solid white;
+			border-left: 10px solid transparent;
+			border-right: 0px solid transparent;
+			z-index: 3;
+		}
+		.receiver::before {
+			content: "";
+			position: absolute;
+			top: -1px;
+			left: -13px;
+			border-top: 11px solid rgb(209 209 209 / 34%);
+			border-right: 0px solid transparent;
+			border-left: 12px solid transparent;
+			filter: blur(0.8px);
+			z-index: 2;
+		}
+		.receiver-name {
+			font-size: 13px;
+			color: blue;
+		}
+
+		.message {
+			overflow-wrap: break-word;
+		}
+
+		.sender .message-timestamp {
+			text-align: end;
+
+			font-size: 11px;
+			color: #e9e9e9;
+		}
+
+		.receiver .message-timestamp {
+			text-align: end;
+
+			font-size: 11px;
+			color: #8c8d8d;
+		}
+
+		.message-settings {
+			opacity: 0;
+			position: absolute;
+			top: 50%;
+			transform: translate(0, -50%);
+
+			background: white;
+			border-radius: 6px;
+			box-shadow: rgba(0, 0, 0, 0.16) 0px 1px 4px;
+		}
+
+		.sender .message-settings {
+			color: black;
+			left: -40px;
+		}
+
+		.receiver .message-settings {
+			right: -40px;
+		}
+
+		.sender:hover .message-settings,
+		.receiver:hover .message-settings {
+			opacity: 1;
+		}
+
+		.message-settings:hover,
+		.menu-options:hover .message-settings {
+			opacity: 1;
+		}
+
+		.menu-options {
+			display: none;
+			top: -3px;
+			width: max-content;
+
+			padding: 5px;
+		}
+
+		.menu-options-section {
+			cursor: pointer;
+			padding: 4px;
+			border-radius: 5px;
+			user-select: none;
+
+			display: flex;
+			align-items: center;
+			gap: 3px;
+		}
+
+		.menu-options-section:hover {
+			background-color: #f5f5f5;
+		}
+
+		.message-settings:hover .menu-options {
+			display: block !important;
+		}
```

Comment:
Gestirei la cosa con javascript togliendo l'hover, in modo che cliccando sui tre punti si apra il menù.